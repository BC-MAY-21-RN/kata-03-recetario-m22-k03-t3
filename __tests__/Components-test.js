import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../src/components/common/card';
import List from '../src/components/common/list';
import Mode from '../src/components/common/mode';
import SearchBar from '../src/components/common/searchBar';
import BarTitleDetail from '../src/components/common/barTitleDetail';

describe('Components', () => {
  it('Card', () => {
    const renderCard = renderer
      .create(
        <Card
          key={1}
          dish={{
            id: 1,
            image:
              'https://img.bestrecipes.com.au/f4jy177E/w643-h428-cfill-q90/br/2020/11/beef-stew-960642-2.jpg',
            title: 'Beef stew',
            servings: 3,
            ingredients: [
              {
                ingredient: 'extra virgin olive oil',
                size: '2 cups',
              },
              {
                ingredient: 'gravy beef, cut into 4cm pieces',
                size: '500 g',
              },
              {
                ingredient: 'large carrots, thickly sliced',
                size: '2',
              },
            ],
          }}
          mode={true}
          isLast={false}
          type={'RECENT'}
          onPress={() => {}}
        />,
      )
      .toJSON();
    expect(renderCard).toMatchSnapshot();
  });
  it('List', () => {
    const renderList = renderer
      .create(
        <List
          type="TRENDING"
          navigation={null}
          mode={true}
          data={[]}
          insertData={x => {}}
        />,
      )
      .toJSON();
    expect(renderList).toMatchSnapshot();
  });
  it('Mode', () => {
    const renderMode = renderer
      .create(<Mode mode={true} changeMode={() => {}} />)
      .toJSON();
    expect(renderMode).toMatchSnapshot();
  });
  it('SearchBar', () => {
    const renderSearchBar = renderer.create(<SearchBar mode={true} />).toJSON();
    expect(renderSearchBar).toMatchSnapshot();
  });
  it('BarTitleDetail', () => {
    const renderBarTitleDetail = renderer
      .create(
        <BarTitleDetail
          navigation={{goBack: () => null}}
          type={'RECENT'}
          name={'frijoles'}
          mode={true}
          dish={{}}
        />,
      )
      .toJSON();
    expect(renderBarTitleDetail).toMatchSnapshot();
  });
});
