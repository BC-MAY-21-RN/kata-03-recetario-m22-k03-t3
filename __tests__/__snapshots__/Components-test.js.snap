// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components BarTitleDetail 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 385,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(43,41,41,0.53)",
        "display": "flex",
        "flexDirection": "column",
        "height": 385,
        "justifyContent": "flex-end",
        "paddingBottom": 42.5,
        "paddingHorizontal": 26.8,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 64,
          "left": 25,
          "position": "absolute",
          "right": 28,
          "top": null,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        < />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 15,
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        < />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "padding": 8,
          }
        }
      >
        < />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "lineHeight": 22,
          "marginBottom": 2.4,
        }
      }
    >
      RECENT
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "lineHeight": 30,
        }
      }
    >
      frijoles
    </Text>
  </View>
</View>
`;

exports[`Components Card 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 300,
      "marginRight": 21.5,
      "opacity": 1,
      "width": 160,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://img.bestrecipes.com.au/f4jy177E/w643-h428-cfill-q90/br/2020/11/beef-stew-960642-2.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 195,
        "width": 160,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 18,
        "marginTop": 13.8,
      }
    }
  >
    Beef stew
  </Text>
</View>
`;

exports[`Components List 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#f91a81",
        "fontSize": 21,
        "marginBottom": 18.48,
      }
    }
  >
    TRENDING
  </Text>
  <RCTScrollView
    horizontal={true}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Components Mode 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 36,
      "marginBottom": 18,
      "marginTop": 22,
      "opacity": 1,
      "width": 142,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4422C5",
        "borderRadius": 40,
        "display": "flex",
        "height": 36,
        "justifyContent": "center",
        "width": 36,
      }
    }
  >
    < />
  </View>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "lineHeight": 18,
        "marginLeft": 14,
      }
    }
  >
    Normal Mode
  </Text>
</View>
`;

exports[`Components SearchBar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(142, 142, 147, 0.12)",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "height": 36,
      "paddingLeft": 8,
      "paddingVertical": 0,
      "width": "100%",
    }
  }
>
  < />
  <TextInput
    placeholder="What do you want to eat?"
    placeholderTextColor="#ffffff"
    style={
      Object {
        "color": "#fff",
        "flex": 1,
        "fontSize": 17,
        "height": "100%",
        "lineHeight": 20,
        "paddingLeft": 8,
        "paddingVertical": 0,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    < />
  </View>
</View>
`;
